src/Common/AssemblyLoadingUtils.cs

```
@@ -0,0 +1,249 @@
+                case "mklimports.dll":
+                case "microsoft.research.controls.decisiontrees.dll":
+                case "microsoft.ml.neuralnetworks.sse.dll":
+                case "neuraltreeevaluator.dll":
+                case "optimizationbuilderdotnet.dll":
+                case "parallelcommunicator.dll":
+                case "microsoft.ml.runtime.runtests.dll":
+                case "scopecompiler.dll":
+                case "tbb.dll":
+                case "internallearnscope.dll":
+                case "unmanagedlib.dll":
+                case "vcclient.dll":
+                case "libxgboost.dll":
+                case "zedgraph.dll":
+                case "__scopecodegen__.dll":
+                case "cosmosClientApi.dll":
+                    return true;
+            }
+
+            foreach (var s in _filePrefixesToAvoid)
+            {
+                if (name.StartsWith(s))
+                    return true;
+            }
+
+            return false;
+        }
+
+        private static void LoadAssembliesInDir(IHostEnvironment env, string dir, bool filter)
+        {
+            if (!Directory.Exists(dir))
+                return;
+
+            // Load all dlls in the given directory.
+            var paths = Directory.EnumerateFiles(dir, "*.dll");
+            foreach (string path in paths)
+            {
+                if (filter && ShouldSkipPath(path))
+                    continue;
+
+                LoadAssembly(env, path);
+            }
+        }
+
+        /// <summary>
+        /// Given an assembly path, load the assembly and register it with the ComponentCatalog.
+        /// </summary>
+        private static Assembly LoadAssembly(IHostEnvironment env, string path)
+        {
+            try
+            {
+                var assembly = Assembly.LoadFrom(path);
+                env.ComponentCatalog.RegisterAssembly(assembly);
+                return assembly;
+            }
+            catch (Exception)
```

Similarly, why do we try to swallow all these errors?

---

I've changed it to only swallow errors from `Assembly.LoadFrom`, and log an error, which is the existing behavior.